checked_message.go

```diff
@@ -41,34 +48,25 @@ type CheckedMessage struct {
 // NewCheckedMessage constructs a CheckedMessage. It's only intended for use by
 // wrapper libraries, and shouldn't be necessary in application code.
 func NewCheckedMessage(logger Logger, lvl Level, msg string) *CheckedMessage {
-	return &CheckedMessage{
-		logger: logger,
-		lvl:    lvl,
-		msg:    msg,
+	for {
+		m := _CMPool.Get().(*CheckedMessage)
+		m.logger, m.lvl, m.msg = logger, lvl, msg
+		return m
 	}
 }
 
-// Write logs the pre-checked message with the supplied fields. It should only
-// be used once; if a CheckedMessage is re-used, it also logs an error message
-// with the underlying logger's DFatal method.
+// Write logs the pre-checked message with the supplied fields. It will call
+// the underlying level method (Debug, Info, Warn, Error, Panic, and Fatal) for
+// the defined levels; the Log method is only called for unknown logging
+// levels.
 //
-// Write will call the underlying level method (Debug, Info, Warn, Error,
-// Panic, and Fatal) for the defined levels; the Log method is only called for
-// unknown logging levels.
+// It MUST be called at most once, since Write will return the *CheckedMessage
+// to an internal pool for potentially immediate re-use.
```

Comment:
Excellent. Perhaps add a scary final sentence explicating the results of re-use - corrupted memory, undefined behavior, race conditions, etc.

Basically, I don't want to assume that all users of the library understand that "returned to an internal pool" means "pending apocalypse and nightmare debugging."